core/src/Streamly/Internal/Data/Array/Foreign.hs

```diff
@@ -213,6 +195,7 @@ producer = Producer step inject extract
 read :: forall m a. (Monad m, Storable a) => Unfold m (Array a) a
 read = Producer.simplify producer
 
+-- XXX This does not exist in mutable arrays
```

Comment:
The comments below anyway say that it is not faster, so we can remove the XXX, because its not a todo.